WireAPI/Sources/WireAPI/APIs/AuthenticationAPI/AuthenticationAPIV0.swift

```diff
@@ -133,4 +133,49 @@ class AuthenticationAPIV0: AuthenticationAPI, VersionedAPI {
     func getDomainRegistration(forEmail email: String) async throws -> DomainRegistrationConfiguration {
         throw AuthenticationAPIError.unsupportedEndpointForAPIVersion
     }
+
+    func validateLoginToken(ssoCode: UUID) async throws {
+        let path = "/sso/initiate-login/\(ssoCode.uuidString)"
+        let request = try URLRequestBuilder(path: path)
+            .withMethod(.head)
+            .build()
+
+        let (_, response) = try await networkService.executeRequest(request)
+
+        return try ResponseParser()
+            .success(code: .ok)
+            .failureSSOError()
+            .parse(code: response.statusCode, data: nil)
+    }
+
+    func getSSOCode() async throws -> UUID? {
+        let path = "/sso/settings"
+        let request = try URLRequestBuilder(path: path)
+            .withMethod(.get)
+            .withAcceptType(.json)
+            .build()
+
+        let (data, response) = try await networkService.executeRequest(request)
+
+        let payload = try ResponseParser()
+            .success(code: .ok, type: SSOSettingsResponseV0.self)
+            .parse(code: response.statusCode, data: data)
+
+        return payload.defaultSSOCode
+    }
+}
+
+private extension ResponseParser {
+
+    func failureSSOError() -> ResponseParser<Success> {
+        var copy = self
+        copy.parseBlocks.append { code, _ in
+            if let error = AuthenticationAPIError.SSOLoginError(responseCode: code) {
+                throw error
+            }
+            return nil
+        }
+        return copy
+    }
+
```

Comment:
suggestion: seeing as this all we need to do is throw an optional error from a code, I would actually suggest to avoid this extension and simply initialize the error inline, i.e don't use the response parser:

```swift
let (_, response) = try await networkService.executeRequest(request)
if let error = AuthenticationAPIError.SSOLoginError(responseCode: response.statusCode) {
  throw error
}
```

Why? Because is simpler and avoids additional boilerplate that won't be reused elsewhere.

Reply:
applied